starboard/android/shared/audio_renderer_passthrough.cc

```diff
@@ -598,6 +592,16 @@ void AudioRendererPassthrough::OnDecoderOutput() {
   auto decoded_audio = decoder_->Read(&decoded_audio_sample_rate);
   SB_DCHECK(decoded_audio);
 
+  // We set |frames_per_input_buffer_| before adding first |decoded_audio| into
+  // |decoded_audios_|. Though |frames_per_input_buffer_| may be used in
+  // UpdateStatusAndWriteData() from another thread, it's ok not to add thread
+  // guard for |frames_per_input_buffer_|.
+  if (frames_per_input_buffer_ == 0) {
+    frames_per_input_buffer_ = ParseAc3SyncframeAudioSampleCount(
+        decoded_audio->data(), decoded_audio->size_in_bytes());
+    SB_LOG(INFO) << "Got frames per input buffer " << frames_per_input_buffer_;
+  }
+
```

Comment:
Consider adding SB_DCHECK(frames_per_input_buffer_ == ParseAc3SyncframeAudioSampleCount(
decoded_audio->data(), decoded_audio->size_in_bytes())) << ...;

So we know if something happens next time.